src/test.rs

```diff
@@ -1,21 +1,127 @@
 #![cfg(test)]
 
 use regex::Regex;
+use std::fs;
+use std::path::Path;
+use std::process::Command;
 use std::str;
 
+fn init_git_repo(path: &Path) {
+    Command::new("git")
+        .current_dir(&path)
+        .arg("init")
+        .output()
+        .unwrap();
+
+    let file = path.join("readme");
+    fs::write(&file, "hello").unwrap();
+
+    Command::new("git")
+        .current_dir(&path)
+        .arg("add")
+        .arg("readme")
+        .output()
+        .unwrap();
+
+    Command::new("git")
+        .current_dir(&path)
+        .arg("commit")
+        .arg("-am")
+        .arg("test")
+        .output()
+        .unwrap();
+}
+
 #[test]
 fn test_init() {
+    let tempdir = tempfile::tempdir().unwrap();
+    let git_dir = tempdir.path();
+
+    init_git_repo(&git_dir);
+
+    let mut out = Vec::new();
+    let res = super::__init(&mut out, &git_dir);
+    assert!(res.is_ok());
+    let out = str::from_utf8(&out).unwrap();
+    let expected = "cargo:rerun-if-changed=.git/index
+cargo:rerun-if-changed=.git/HEAD
+cargo:rerun-if-changed=.git/refs
+cargo:rustc-env=GIT_REVISION=[0-9a-f]+";
+    println!("{out}");
+    println!("{expected}");
+    assert!(Regex::new(expected).unwrap().is_match(out));
+}
+
+#[test]
+fn test_init_subdir() {
+    let tempdir = tempfile::tempdir().unwrap();
+    let git_dir = tempdir.path();
+
+    init_git_repo(&git_dir);
+
+    let manifest_dir = git_dir.join("subdir");
+    std::fs::create_dir(&manifest_dir).unwrap();
+
     let mut out = Vec::new();
-    let res = super::__init(&mut out);
+    let res = super::__init(&mut out, &manifest_dir);
     assert!(res.is_ok());
     let out = str::from_utf8(&out).unwrap();
+    let expected = &format!(
+        "cargo:rerun-if-changed={gd}/.git/index
+cargo:rerun-if-changed={gd}/.git/HEAD
+cargo:rerun-if-changed={gd}/.git/refs
+cargo:rustc-env=GIT_REVISION=[0-9a-f]+",
+        gd = git_dir.display()
+    );
     println!("{out}");
-    assert!(Regex::new(
-        "cargo:rerun-if-changed=.git/index
+    println!("{expected}");
+    assert!(Regex::new(expected).unwrap().is_match(out));
```

Comment:
This test is failing for me on a mac because out and expected are:

Out:
```
cargo:rerun-if-changed=/private/var/folders/75/wqcxsy716bq0qht_8jtvw85h0000gn/T/.tmpGyGjKM/.git/index
cargo:rerun-if-changed=/private/var/folders/75/wqcxsy716bq0qht_8jtvw85h0000gn/T/.tmpGyGjKM/.git/HEAD
cargo:rerun-if-changed=/private/var/folders/75/wqcxsy716bq0qht_8jtvw85h0000gn/T/.tmpGyGjKM/.git/refs
cargo:rustc-env=GIT_REVISION=ea8ad1945aea4b162e58ebbb0e964442842b68cc
```

Expected:
```
cargo:rerun-if-changed=/var/folders/75/wqcxsy716bq0qht_8jtvw85h0000gn/T/.tmpGyGjKM/.git/index
cargo:rerun-if-changed=/var/folders/75/wqcxsy716bq0qht_8jtvw85h0000gn/T/.tmpGyGjKM/.git/HEAD
cargo:rerun-if-changed=/var/folders/75/wqcxsy716bq0qht_8jtvw85h0000gn/T/.tmpGyGjKM/.git/refs
cargo:rustc-env=GIT_REVISION=[0-9a-f]+
```

On mac `/var` is a symlink to `/private/var`.

Reply:
The tests are also failing on ci, unable to find the git directory:

```
cargo:rerun-if-changed=.git/index
cargo:rerun-if-changed=.git/HEAD
cargo:rerun-if-changed=.git/refs
cargo:rustc-env=GIT_REVISION=

cargo:rerun-if-changed=.git/index
cargo:rerun-if-changed=.git/HEAD
cargo:rerun-if-changed=.git/refs
cargo:rustc-env=GIT_REVISION=[0-9a-f]+
```

Reply:
Is that failure on ci a new line difference on the end of one?